release-notes/swan-lake-2201.1.0-release-note.md

```diff
@@ -30,9 +30,364 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
+
+```ballerina
+public function main() {
+    io:println("abc".padStart(5, "#")); // "##abc"
+    io:println("abc".padStart(5)); // "  abc"
+
+    io:println("abc".padEnd(5, "#")); // "abc##"
+    io:println("abc".padEnd(5)); // "abc  "
+
+    io:println("123".padZero(5)); // "00123"
+    io:println("123".padZero(5, "#")); // "##123"
+}
+```
+
+#### Improvements
+
+##### Disallow inferring array length in contexts that are not permitted
+
+Inferring array length has been restricted to list constructors in variable and constant declarations. Moreover, only the first dimension can be inferred in multidimensional arrays.
+
+```ballerina
+int[*] x1 = [1, 2]; // Supported.
+
+int[2] y = [1, 2];
+int[*] x2 = y; // Not supported. Requires a list constructor to infer the array length.
+
+int[*][2] x3 = [[1, 2], [1, 2]]; // Supported.
+int[*][*] x4 = [[1, 2], [1, 2]]; // Not supported. Only the first dimension can be inferred.
+```
+
+##### Revamped `lang.float:round` langlib function
+
+The function signature has been changed to have an extra argument `fractionDigits`, where the user can choose the number of fraction digits of the rounded result. When `fractionDigits` is zero, the function rounds to an integer.
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    float x = 555.545;
+    float y = 5.5565;
+    int fractionDigits = 3;
+
+    io:println(555.545.round(1)); // 555.5
+    io:println(555.545.round(2)); // 555.54
+    io:println(float:round(x)); // 556.0
+    io:println(float:round(x, fractionDigits = 0)); // 556.0
+    io:println(float:round(x, 1));  // 555.5
+    io:println(y.round(2)); // 5.56
+    io:println(y.round(fractionDigits)); // 5.556
+}
+```
+
+##### Revamped `lang.decimal:round` langlib function
+
+The function signature has been changed to have an extra argument `fractionDigits`, where the user can choose the number of fraction digits of the rounded result. When `fractionDigits` is zero, the function rounds to an integer.
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    io:println(5.55.round(1)); // 5.6
+    decimal x = 5.55;
+    io:println(decimal:round(x)); // 6
+    io:println(decimal:round(5.55, fractionDigits = 0)); // 6
+    io:println(decimal:round(5.5565, fractionDigits = 3)); // 5.556
+}
+```
+
+##### An unreachable panic statement no longer results in a compilation error
+
+```ballerina
+function fn() returns string {
+    int|string a = 10;
+
+    if a is int {
+        return "INT";
+    } else {
+        return "STRING";
+    }
+
+    panic error("Not Reached!"); // unreachable, but not an error.
+}
+```
+
+##### `error:Clonable` Cloneable definition in error module is now public
```

Comment:
```suggestion
##### Updated public `error:Clonable` cloneable definition in error module

The `error:Clonable` cloneable definition in the error module is now public.
```
It is not a good practice to have sentences on topics. It would be in noun or verb form.

Reply:
@praneesha Can we say "Updated error:Cloneable to be public in error module" ?

Reply:
+1